### DISCUSSION TITLE
Query for retrieving "not disabled" users in Active Directory

### DISCUSSION THREAD

**Discussion by emiliosh** [DirectoryTree/LdapRecord#651] [Q&A] [answered]
Hi, I'm struggling in getting only enabled Active Directory users.

I'm using an Scope like this.

```
public function apply(Builder $query, Model $model): void
{
   $accountcontrol = new AccountControl(AccountControl::ACCOUNTDISABLE);
   $query->in('ou=_Usuarios,{base}')->whereHas('employeeid')->rawFilter('!'.$accountcontrol->filter());
}
```
It returns:

   **LdapRecord\LdapRecordException
  ldap_search(): Search: Bad search filter**

If I remove '!'. concat everything it's ok, but I need it with !.

Using LdapRecord v3.

Thanks in advance.

**Comment by stevebauman** [comment, marked as answer]
Hi @emiliosh,

If you'd like to use it, there is already a built-in query builder method to return only enabled users:

https://github.com/DirectoryTree/LdapRecord/blob/496fd4496be081f4142c38fb39546e2357c349c6/src/Query/Model/ActiveDirectoryBuilder.php#L37-L45

```php
public function apply(Builder $query, Model $model): void
{
    $query->whereEnabled();
}
```

However, if you'd like to do this yourself, try:

```php
$query->notFilter(
    fn ($query) => $query->rawFilter(
        (new AccountControl)->setAccountIsDisabled()->filter()
    )
);
```

Let me know if this answers your question! 👍

**Reply by emiliosh** [reply]
Thanks a lot, works perfectly!
